aten/src/ATen/native/cuda/BatchLinearAlgebraLib.cu

```diff
@@ -36,6 +36,40 @@ static Tensor get_device_pointers(const Tensor& input) {
       input.options().dtype(at::kLong));
 }
 
+template <typename scalar_t>
+void apply_geqrf_batched(const Tensor& input, const Tensor& tau, int64_t m64, int64_t n64) {
+// AMD ROCm backend is implemented via rewriting all CUDA calls to HIP
+// rocBLAS does not implement BLAS-like extensions of cuBLAS, they're in rocSOLVER
+// rocSOLVER is currently not used in ATen, therefore we raise an error in this case
+#ifndef CUDART_VERSION
+  TORCH_CHECK(false, "geqrf: Batched version is supported only with cuBLAS backend.")
+#else
+  auto batch_size = cuda_int_cast(batchCount(input), "batch_size");
+  auto m = cuda_int_cast(m64, "m");
+  auto n = cuda_int_cast(n64, "n");
+  auto lda = std::max<int>(1, m);
+
+  // cuBLAS batched geqrf requires input to be the device array of pointers to device single matrices
+  Tensor input_ptr_array = get_device_pointers<scalar_t>(input);
+  Tensor tau_ptr_array = get_device_pointers<scalar_t>(tau.unsqueeze(-1));
```

Comment:
Hmm, why is there an `unsqueeze` call?

Reply:
That's a workaround as `get_device_pointers` works only for array of matrices and `tau` is a vector.